src/components/protocol_handler/src/handshake_handler.cc

```diff
@@ -0,0 +1,197 @@
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "protocol_handler/handshake_handler.h"
+
+#include <bson_object.h>
+
+#include "protocol_handler/protocol_handler_impl.h"
+#include "protocol_handler/protocol_packet.h"
+#include "security_manager/security_manager.h"
+#include "protocol_handler/session_observer.h"
+
+namespace protocol_handler {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "ProtocolHandler")
+
+// DEPRECATED
+HandshakeHandler::HandshakeHandler(
+    ProtocolHandlerImpl& protocol_handler,
+    SessionObserver& session_observer,
+    uint32_t connection_key,
+    ConnectionID connection_id,
+    uint8_t session_id,
+    uint8_t protocol_version,
+    uint32_t hash_id,
+    ServiceType service_type,
+    const std::vector<int>& force_protected_service,
+    const bool is_new_service,
+    ProtocolPacket::ProtocolVersion& full_version,
+    std::shared_ptr<uint8_t> payload)
+    : protocol_handler_(protocol_handler)
+    , session_observer_(session_observer)
+    , context_()
+    , full_version_(full_version)
+    , protocol_version_(protocol_version)
+    , payload_(payload) {}
+
+HandshakeHandler::HandshakeHandler(
+    ProtocolHandlerImpl& protocol_handler,
+    SessionObserver& session_observer,
+    ProtocolPacket::ProtocolVersion& full_version,
+    const StartingSessionContext& context,
+    const uint8_t protocol_version,
+    std::shared_ptr<uint8_t> payload)
+    : protocol_handler_(protocol_handler)
+    , session_observer_(session_observer)
+    , context_(context)
+    , full_version_(full_version)
+    , protocol_version_(protocol_version)
+    , payload_(payload) {}
+
+HandshakeHandler::~HandshakeHandler() {
+  LOG4CXX_DEBUG(logger_, "Destroying StartSessionHandler: " << this);
```

Comment:
@AKalinich-Luxoft is it correct name of class?

Reply:
@LuxoftAKutsan no, this is an old debug message. Fixed in https://github.com/smartdevicelink/sdl_core/pull/1971/commits/a87c65251e81cb914deaa1f0fe1716175fe1270d